src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxTreeFactoryService.cs

```diff
@@ -48,8 +48,15 @@ public override SyntaxNode DeserializeNodeFrom(Stream stream, CancellationToken
                 return CSharpSyntaxNode.DeserializeFrom(stream, cancellationToken);
             }
 
+            public override bool CanCreateRecoverableTree(SyntaxNode root)
+            {
+                var cu = root as CompilationUnitSyntax;
+                return base.CanCreateRecoverableTree(root) && cu != null && cu.AttributeLists.Count == 0;
```

Comment:
Also, the cu != null check can probably happen before the base.CanCreateRecoverableTree(...) clause. Might be needed for correctness if a tree of a different language somehow flowed through here.